### PR TITLE
Update Inversify to latest

### PR DESCRIPTION
**Pull request by msujew**
#### What it does

Closes https://github.com/eclipse-theia/theia/issues/14431.

The latest minor version of Inversify (6.1.0) introduced some behavior changes with regards to injected base classes. Inversify validates that the constructor arguments are in some way satisfied (by asserting constructor function length). Since we sometimes use an inheritance pattern that breaks this assertion, this leads to runtime failures in newer versions of Inversify.

This change adds `@unmanaged()` decorations to all constructors that aren't injected. Also adds a few more type constraints during compile time where Inversify got more strict in its typing.

#### How to test

Assert that the CI is green, and the application builds and runs without any start error.

#### Follow-ups

I couldn't really find *all* places where we're using the above mentioned pattern, as I couldn't find a way to easily search for it.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
     "http-proxy-agent": "^5.0.0",
     "https-proxy-agent": "^5.0.0",
     "iconv-lite": "^0.6.0",
-    "inversify": "^6.0.1",
```
Comment (sdirix) [inline packages/core/package.json:54]: Inversify `6.1.3` was just released, containing the build fix. So we should already indicate that we want to use that.

Comment (msujew) [reply packages/core/package.json:54]: Yeah, good idea.

**Review by sdirix — COMMENTED** [review]
I was able to build and start the browser application and did not encounter any error just by clicking a bit around and trying some features.

I then checked with the regex `@injectable(.|\n)*?constructor` to find all places where an `@unmanaged` might be needed.

I found the following `@injectable` classes which either have constructor parameters which are neither injected nor are declared unmanaged or execute a `super` call to a parent class without such declarations for their parameters.

- [AbstractDialog](https://github.com/eclipse-theia/theia/blob/1900a805a877e7bee9d87e228036060b8d91b233/packages/core/src/browser/dialogs.ts#L154-L155)
- [AbstractTreeDecoratorService](https://github.com/eclipse-theia/theia/blob/1900a805a877e7bee9d87e228036060b8d91b233/packages/core/src/browser/tree/tree-decorator.ts#L173)
- [BaseWidget](https://github.com/eclipse-theia/theia/blob/1900a805a877e7bee9d87e228036060b8d91b233/packages/core/src/browser/widgets/widget.ts#L120-L122)
- [BlameDecorator](https://github.com/eclipse-theia/theia/blob/1900a805a877e7bee9d87e228036060b8d91b233/packages/git/src/browser/blame/blame-decorator.ts#L31)
- [MarkerTree](https://github.com/eclipse-theia/theia/blob/1900a805a877e7bee9d87e228036060b8d91b233/packages/markers/src/browser/marker-tree.ts#L35-L36)
- [HostedPlugin](https://github.com/eclipse-theia/theia/blob/1900a805a877e7bee9d87e228036060b8d91b233/packages/plugin-ext/src/hosted/common/hosted-plugin.ts#L106)
- [ProxyPluginDeployerEntry](https://github.com/eclipse-theia/theia/blob/1900a805a877e7bee9d87e228036060b8d91b233/packages/plugin-ext/src/main/node/plugin-deployer-proxy-entry-impl.ts#L28)
- [AbstractHostedPluginSupport](https://github.com/eclipse-theia/theia/blob/1900a805a877e7bee9d87e228036060b8d91b233/packages/plugin-ext/src/hosted/common/hosted-plugin.ts#L106)
- [Process](https://github.com/eclipse-theia/theia/blob/1900a805a877e7bee9d87e228036060b8d91b233/packages/process/src/node/process.ts#L89-L92)
- [Task](https://github.com/eclipse-theia/theia/blob/1900a805a877e7bee9d87e228036060b8d91b233/packages/task/src/node/task.ts#L46-L48)
- [DefaultTerminalProfileService](https://github.com/eclipse-theia/theia/blob/1900a805a877e7bee9d87e228036060b8d91b233/packages/terminal/src/browser/terminal-profile-service.ts#L98)

**Review by sdirix — APPROVED** [review]
Works for me! Thanks for the work!

**Comment by tsmaeder** [comment]
What's keeping this PR from being merged?

**Pull request merged**